### PR TITLE
HTTP2: Fix HttpRequestHeaders no longer being pooled

### PR DESCRIPTION
**Pull request by JamesNK**
HttpRequestHeaders is showing up in profile:

image

I believe there was a regression caused when Latin1 headers https://github.com/dotnet/aspnetcore/pull/18255 was merged with Justin's stream pooling changes.

### CODE REVIEW

**Comment by jkotalik** [comment]
I opened https://github.com/dotnet/aspnetcore/pull/19312 already, but this looks a bit more complete.

**Review by jkotalik — APPROVED** [review]

File: src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs [hunk 1]
```diff
@@ -6105,7 +6105,7 @@ public unsafe void Append(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
                 }
 
                 // We didn't have a previous matching header value, or have already added a header, so get the string for this value.
-                var valueStr = value.GetRequestHeaderStringNonNullCharacters(_useLatin1);
+                var valueStr = value.GetRequestHeaderStringNonNullCharacters(UseLatin1);
```
Comment (gfoidl) [inline src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs]: This is a `*.Generated.cs`-file, but I don't see any change in the generator itself. So this change will be overwritten soon.

Maybe I'm missing something...

Comment (gfoidl) [reply src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs]: This line https://github.com/dotnet/aspnetcore/blob/3787d7e7f070543cc9368d589a504fa8c4bd4830/src/Servers/Kestrel/shared/KnownHeaders.cs#L989 should be updated.

Comment (gfoidl) [reply src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs]: Ohhhhhh, please ignore my comments. It's already there.

> (not so easy to find...)

Led me to this comments. Sorry!

**Review by halter73 — APPROVED** [review]

**Comment by mjsabby** [comment]
This change (in addition to the one already committed) will make it in 3.1.3 correct?

**Comment by JamesNK** [comment]
No. This was a regression to new functionality on master when the 3.1.3 branch was merged back.

**Pull request merged**